src/HeatMap/components/HeatMapContainer.ts

```diff
@@ -39,13 +40,29 @@ export default class HeatMapContainer extends Component<HeatMapContainerProps, H
         this.resetSubscriptions(newProps.mxObject);
         if (!this.state.alertMessage) {
             this.fetchData(newProps.mxObject);
+            if (newProps.mxObject) {
+                this.setRefreshInterval(newProps.refreshInterval);
+            }
         }
     }
 
     componentWillUnmount() {
         if (this.subscriptionHandle) {
             window.mx.data.unsubscribe(this.subscriptionHandle);
         }
+        if (this.intervalID) {
+            clearInterval(this.intervalID);
+        }
+    }
+
+    private setRefreshInterval(refreshInterval: number) {
+        if (refreshInterval > 0) {
```

Comment:
This function will be called every time the properties are changing. So multiple interfals will be started, overwriting the last ID. causing in the un-mount not be able to clear the interval. causes the internal to function will the component no longer exists.

``` ts
if (this.intervalID) {
    window.clearInterval(this.intervalID);
}
```

Also move the `newProps.mxObject` check to this function. So if the mxobject not available the interval should be cleared.

Reply:
There is a call for compountWillUnmount in the reset subscriptions... So the clear will be done before another is set